fendermint/vm/actor_interface/src/init.rs

```diff
@@ -20,9 +21,22 @@ define_singleton!(INIT { id: 1, code_id: 2 });
 pub type AddressMap = HashMap<Address, ActorID>;
 
 /// Delegated address of an Ethereum built-in actor.
-fn eth_builtin_deleg_addr(id: ActorID) -> Address {
+///
+/// This is based to what seems to be going on in the `CREATE_EXTERNAL` method
+/// of the EAM actor when it determines a robust address for an account ID,
+/// in that we take something known (a public key, or in this case the ID),
+/// hash it and truncate the results to 20 bytes.
+///
+/// But it's not a general rule of turning actor IDs into Ethereum addresses!
+/// It's just something we do to assign an address that looks like an Ethereum one.
+pub fn builtin_actor_eth_addr(id: ActorID) -> EthAddress {
```

Comment:
So let me see if I understand, the idea is that: 
* For the EVM contracts initialized in genesis we choose an ID address (like `64` for the gateway). 
* Instead of using the masked ID to represent its Ethereum address, we use this `hash20` trick, that way we prevent the issue we were facing of `address is not a delegated address`. 

This means that we should change the default Ethereum address of the gateway and registry contracts that we point to now in child subnets?

Reply:
Exactly, somewhere something would freak out if id did not have a delegated address that isn't a masked ID address. I forgot where that check is and I can't find it now :( 

We use this trick based on what the EAM actor does for `Account` types, hashing their ID and calling it a _robust address_. 

These addresses are printed in the log when you run the node. I don't know them by heart, but luckily the smoke tests print the logs on CI, so we can look at the [end to end output](https://github.com/consensus-shipyard/fendermint/actions/runs/6660009863/job/18100303444?pr=338): 

```console
2023-10-26T21:51:42.984013Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=107 eth_addr=0x8c3fb5977c07ef1f42e76eca5b7895282b5639c4 id_addr=0xff0000000000000000000000000000000000006b fqn="src/lib/AccountHelper.sol:AccountHelper"
2023-10-26T21:51:42.984529Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=108 eth_addr=0xee239fddbca8377611ee9f15a6a287c277b52b8c id_addr=0xff0000000000000000000000000000000000006c fqn="src/lib/CheckpointHelper.sol:CheckpointHelper"
2023-10-26T21:51:42.986787Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=109 eth_addr=0x4825cbdeb7f6abe08df977b0f03590cdd2918639 id_addr=0xff0000000000000000000000000000000000006d fqn="src/lib/LibStaking.sol:LibStaking"
2023-10-26T21:51:42.988581Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=110 eth_addr=0x62b6272305ffbffec85cbffa22b6a67f229d1976 id_addr=0xff0000000000000000000000000000000000006e fqn="SubnetActorManagerFacet.sol:SubnetActorManagerFacet"
2023-10-26T21:51:42.989722Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=111 eth_addr=0xd8aee27985d849e01a2f2f75409c9011143a19a1 id_addr=0xff0000000000000000000000000000000000006f fqn="src/lib/SubnetIDHelper.sol:SubnetIDHelper"
2023-10-26T21:51:42.991031Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=112 eth_addr=0x5e1cb859c144004d331b161cebf2c789a17274e5 id_addr=0xff00000000000000000000000000000000000070 fqn="GatewayGetterFacet.sol:GatewayGetterFacet"
2023-10-26T21:51:42.991912Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=113 eth_addr=0x92c6b38935c75592f198f5bb2da741cb9fc04317 id_addr=0xff00000000000000000000000000000000000071 fqn="SubnetActorGetterFacet.sol:SubnetActorGetterFacet"
2023-10-26T21:51:42.992957Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=114 eth_addr=0x711899f0224604b34098d4c620ddc8e07fe55adc id_addr=0xff00000000000000000000000000000000000072 fqn="src/lib/CrossMsgHelper.sol:CrossMsgHelper"
2023-10-26T21:51:42.993979Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=115 eth_addr=0x5c7d6cece8552eaa20f928ed1b7acf4f0448535f id_addr=0xff00000000000000000000000000000000000073 fqn="GatewayManagerFacet.sol:GatewayManagerFacet"
2023-10-26T21:51:42.994534Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=116 eth_addr=0xf3dda05200d10a38782b121dd440b136bff17c36 id_addr=0xff00000000000000000000000000000000000074 fqn="src/lib/StorableMsgHelper.sol:StorableMsgHelper"
2023-10-26T21:51:42.995428Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=117 eth_addr=0x2876e69b1cf47d1fa467ec3beccee43e6eae7c02 id_addr=0xff00000000000000000000000000000000000075 fqn="GatewayMessengerFacet.sol:GatewayMessengerFacet"
2023-10-26T21:51:42.998840Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum library actor_id=118 eth_addr=0x6fd99fcafbba6def5b13c0239921dd269b6762d8 id_addr=0xff00000000000000000000000000000000000076 fqn="GatewayRouterFacet.sol:GatewayRouterFacet"
2023-10-26T21:51:43.005347Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum contract actor_id=64 eth_addr=0x77aa40b105843728088c0132e43fc44348881da8 id_addr=0xff00000000000000000000000000000000000040 contract_name="GatewayDiamond"
2023-10-26T21:51:43.006893Z  INFO fendermint_vm_interpreter::fvm::genesis: deployed Ethereum contract actor_id=65 eth_addr=0x74539671a1d2f1c8f200826baba665179f53a1b7 id_addr=0xff00000000000000000000000000000000000041 contract_name="SubnetRegistry"
2023-10-26T21:51:43.007501Z  INFO fendermint_app::app: init chain state_root="bafy2bzacebzqdhhhasvwtnfagra2xepfi4ruwmgmvhtp4n2hlfpmfui2cjg22" app_hash="0171A0E4022017E64DE6C5C4F458B8DAD362B62689D7D5C5067584A292A23E05C381753E4CAC"
```